docs/user/troubleshooting.asciidoc

```diff
@@ -68,3 +68,32 @@ Open Kibana Logs and search for the HTTP server response for the endpoint. It lo
+
+== Trace {es} query to the origin in {kib}
+Sometimes the {es} server might be slowed down by the execution of an expensive query. Such queries are logged to {es}'s https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules-slowlog.html#search-slow-log[Search Slow Log] file. But there is a problem: it's impossible to say who triggered a slow search request - a {kib} instance or a user accessing an {es} endpoint directly.
+To simplify the investigation of such cases, https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules-slowlog.html#search-slow-log[Search Slow Log] file includes the `x-opaque-id` header that might provide additional information about a request if it originated from {kib}.
+
+WARNING: At the moment, {kib} can only highlight cases where a slow query originated from a {kib} visualization, Lens, Discover, or {kib} Alerting.
+
+For example, if a request to {es} was initiated by a Vega visualization on a Dashboard, you will see in the slow logs:
+`"souce": { "id": "c89d1ab3-b4a7-4920-a64a-22a910a413b0;kibana:application:dashboard:edf84fe0-e1a0-11e7-b6d5-4dc382ef7f5b;visualization:Vega:cb099a20-ea66-11eb-9425-113343a037e3" }`
+
+Let’s look closer at the format of the string.
+The id value starts with  `c89d1ab3-b4a7-4920-a64a-22a910a413b0`, which is a unique identifier of a request set by the Kibana server.
+The part after the `kibana` prefix tells us that the request was triggered by Dashboard with id `edf84fe0-e1a0-11e7-b6d5-4dc382ef7f5b` and Vega visualisation with id `cb099a20-ea66-11eb-9425-113343a037e3`.
+
+If the provided information is not enough to identify a visualization to adjust its parameters, you can configure {kib} logs to provide a human-readable description and a link to a source of the request:
+[source,yml]
+----
+logging:
+  loggers:
+    - name: execution_context
+      level: debug
+      appenders: [console]
+----
+Now, you can see the request to {es} has been initiated by `[Logs] Unique Visitor Heatmap` visualization embedded in `[Logs] Web Traffic` dashboard. You can navigate to the provided URL to change some parameters of the visualization.
+
+[source,text]
+----
+[DEBUG][execution_context] stored the execution context: {"parent":{"type":"application","name":"dashboard","id":"edf84fe0-e1a0-11e7-b6d5-4dc382ef7f5b","description":"[Logs] Web Traffic","url":"/view/edf84fe0-e1a0-11e7-b6d5-4dc382ef7f5b"},"type":"visualization","name":"Vega","id":"cb099a20-ea66-11eb-9425-113343a037e3","description":"[Logs] Unique Visitor Heatmap","url":"/app/visualize#/edit/cb099a20-ea66-11eb-9425-113343a037e3"}
```

Comment:
Can you put line breaks in this so it is easier to read?

Reply:
I can, however, we don't format log output